lib/credentials/shared_ini_file_credentials.js

```diff
@@ -105,9 +109,14 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
         return;
       }
 
-      this.accessKeyId = profile['aws_access_key_id'];
-      this.secretAccessKey = profile['aws_secret_access_key'];
-      this.sessionToken = profile['aws_session_token'];
+      if (this.useCredentialProcess) {
```

Comment:
Since `useCredentialProcess` is set to `true` in the default chain, this is still incompatible with the CLI. Given a profile with like the following:
```
[foo]
credential_process = /path/to/bin --flag
aws_access_key_id = asdf
aws_secret_access_key = hjkl
```

the SDK for JS would execute `/path/to/bin --flag`, whereas the AWS CLI would not.

Rather than adding a constructor flag, this could be made compatible by changing the order of the branches in the if statement:

```javascript
if (profile['aws_access_key_id'] || profile['aws_secret_access_key']) {
    // use the profile as static creds
} else if (profile['credential_process']) {
    // execute the credential process
}
```

Reply:
I am not sure that is true given that I added an additional instance to the defaultProvider list, so the instance without passing useCredentialProcess will execute first, and if the credentials do not exist in the ini then that instance will error out, and go to the instance in the chain with it set to true.

The example you provided for should be handled in one of the test cases I added to the credential_provider_chain.spec.js

`AWS.CredentialProviderChain.defaultProviders = [
  function () { return new AWS.EnvironmentCredentials('AWS'); },
  function () { return new AWS.EnvironmentCredentials('AMAZON'); },
  function () { return new AWS.SharedIniFileCredentials(); },
  function () { return new AWS.SharedIniFileCredentials({ useCredentialProcess: true }); },
  function () {
    if (AWS.ECSCredentials.prototype.isConfiguredForEcsCredentials()) {
      return new AWS.ECSCredentials();
    }
    return new AWS.EC2MetadataCredentials();
  }
];`

That said, I will test it out tonight and make the adjustments you mentioned if it is necessary.

Reply:
Ah my mistake; I misread the change to the default chain and missed that it was an addition rather than an update.

In any case, just changing the order of the `if/else` block seems simpler and has fewer moving parts.

Reply:
Fair enough. 
I had tried to mimic the example you had cited from boto where they " SharedCredentialProvider" followed by "ProcessProvider" in the providers list. I thought that's what you were originally asking for, but if preferred I can switch it to the simpler implementation you mentioned.